frontend/src/components/Profile/Profile.js

```diff
@@ -16,30 +15,50 @@ export default function Profile(props) {
     const [numFollowings, setNumFollowings] = useState(undefined);
     const [numPictures, setNumPictures] = useState(undefined);
 
+    const [refreshToggle, setRefreshToggle] = useState(true);
+    const [currentClient, setCurrentClient] = useState(undefined);
+    const history = useHistory();
     useEffect(() => {
         const loadProfile = () => {
             axios.get(global.config.BACKEND_URL + `/account/profile/${props.username}`)
-            .then(
-                (response) => {
-                    setUsername(response.data.username);
-                    setBiography(response.data.biography);
-                    setDisplayName(response.data.displayName);
-                    setNumFollowers(response.data.numFollowers);
-                    setNumFollowings(response.data.numFollowings);
-                    setNumPictures(response.data.numPictures);
-            })
-            .catch((error) => {
-                console.log(error.response);
-                if (error.response && error.response.data && error.response.data.message) {
-                    setErrorMsg(error.response.data.message);
-                }
-                else {
-                    setErrorMsg("An unknown error occurred.");
-                }
-            })
+                .then(
+                    (response) => {
+                        setUsername(response.data.username);
+                        setBiography(response.data.biography);
+                        setDisplayName(response.data.displayName);
+                        setNumFollowers(response.data.numFollowers);
+                        setNumFollowings(response.data.numFollowings);
+                        setNumPictures(response.data.numPictures);
+                    })
+                .catch((error) => {
+                    console.log(error.response);
+                    if (error.response && error.response.data && error.response.data.message) {
+                        setErrorMsg(error.response.data.message);
+                    }
+                    else {
+                        setErrorMsg("An unknown error occurred.");
+                    }
+                })
         }
+
+        const getCurentClient = () => {
+            axios.get(global.config.BACKEND_URL + `/account`)
+                .then((response) => {
+                    setCurrentClient(response.data.username);
+                }).catch(() => {
+                    alert("Unexpected error, redirecting to home");
+                    history.push("/");
+                });
+        }
+
         loadProfile();
-    }, [props])
+        getCurentClient();
+    }, [props, refreshToggle, history])
+
+    const handleFollowChange = () => {
+        // Re-render jsx, when FollowingButton has trigger a follow or un-follow
```

Comment:
Don't know how pertinent this comment is...

Reply:
lol I'll remove it... thought it could be useful for those who read the code as the function is being called from a child component